adoc/chapters/programming_interface.adoc

```diff
@@ -105,6 +105,12 @@ where [code]#<backend_name># is the name of the <<backend>> as defined in the
 namespaces for any extensions that a vendor may provide, including any
 <<backend>> that the vendor may define outside of the Khronos SYCL group.
 
+An end-user should not declare anything in the [code]#::sycl#
+namespace or its nested namespaces, except if allowed for specific
+cases by the specification itself (like customization points in some
+extensions for example).
+
+
```

Comment:
The C++ spec uses this wording:

> Unless otherwise specified, the behavior of a C++ program is undefined if it adds declarations or definitions to namespace std or to a namespace within namespace std[.](http://eel.is/c++draft/namespace.std#1.sentence-1)